application/models/message.go

```diff
@@ -83,3 +88,43 @@ func (m *Message) GetThread(requestFields []string) (*Thread, error) {
 	}
 	return &thread, nil
 }
+
+// Create a new message. Sends an `EventApiMessageCreated` event.
+func (m *Message) Create() error {
+	if err := DB.Create(m); err != nil {
+		return err
+	}
+
+	if err := DB.Load(m, "SentBy", "Thread"); err != nil {
+		return err
+	}
+
+	if err := DB.Load(&m.Thread, "Participants", "Post"); err != nil {
+		return err
+	}
+
+	uiUrl := envy.Get(domain.UIURLEnv, "")
+	for _, tp := range m.Thread.Participants {
+		if tp.ID == m.SentBy.ID {
+			continue
+		}
+		e := events.Event{
+			Kind:    domain.EventApiMessageCreated,
+			Message: "New Message from " + m.SentBy.Nickname,
+			Payload: events.Payload{
+				"toName":         tp.Nickname,
```

Comment:
Instead of all these string keys, how about doing something like this (assuming you put the struct outside the function.  I added a comment in the listeners about what to do there.)  ...
```
	type MessageCreatedEventData struct {
		MessageCreatorNickName string
		MessageCreatorEmail    string
		MessageContent         string
		PostUUID               string
		PostTitle              string
		ThreadUUID             string
		ThreadParticipants     []struct{ Nickname, Email string }
	}

	eventData := MessageCreatedEventData{
		MessageCreatorNickName: m.SentBy.Nickname,
		MessageCreatorEmail:    m.SentBy.Email,
		MessageContent:         m.Content,
		PostUUID:               m.Thread.Post.Uuid.String(),
		PostTitle:              m.Thread.Post.Title,
		ThreadUUID:             m.Thread.Uuid.String(),
	}

	for _, tp := range m.Thread.Participants {
		if tp.ID == m.SentBy.ID {
			continue
		}
		eventData.ThreadParticipants = append(eventData.ThreadParticipants,
			struct{ Nickname, Email string }{Nickname: tp.Nickname, Email: tp.Email})
	}

	e := events.Event{
		Kind:    domain.EventApiMessageCreated,
		Message: "New Message from " + m.SentBy.Nickname,
		Payload: events.Payload{
			"eventData": eventData,
		},
	}

	emitEvent(e)
```